### ISSUE TITLE
execClient does not permit returning results

### ISSUE THREAD

**Issue by iphydf** [msgpack/msgpack-haskell#71]
`execClient` returns `IO ()`, so it always returns unit, preventing me from turning a `Client a` into an `IO a`. I'd like to be able to call an rpc method and then process the result further after tearing down the TCP connection without staying inside the Client monad.

**Comment by iphydf** [comment]
FYI: https://hackage.haskell.org/package/network-msgpack-rpc-0.0.5/docs/Network-MessagePack-Client.html#v:execClient implemented this.

**Comment by hvr** [comment]
@iphydf yeah, I do intend to implement this for the next major release of msgpack-rpc (this is a breaking change after all)
